include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,310 @@
+                                 uint32_t                                              plugin_abi,
+                                 const char                                           *plugin_id);
+
+   // Sets the collection to which the preset belongs to.
+   void(CLAP_ABI *set_collection_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                     const char *collection_id);
+
+   // Sets the flags, see clap_preset_discovery_flags.
+   // If unset, they are then inherited from the location.
+   void(CLAP_ABI *set_flags)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                             uint32_t                                              flags);
+
+   // Sets the preset name.
+   // The preset name can be infered from the file name, but it is mandatory to set it for
+   // container's preset.
+   void(CLAP_ABI *set_name)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                            const char                                           *name);
+
+   // Adds a creator name for the preset.
+   void(CLAP_ABI *add_creator)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *creator);
+
+   // Sets a description of the preset.
+   void(CLAP_ABI *set_description)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                   const char *description);
+
+   // Sets the creation time and last modification time of the preset.
+   // The timestamps are in seconds since EPOCH.
+   // If one of the time isn't known, then set it to 0.
+   // If this function is not called, then the indexer may look at the file's creation and
+   // modification time.
+   void(CLAP_ABI *set_timestamps)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  uint64_t creation_time,
+                                  uint64_t modification_time);
+
+   // Adds a feature to the preset.
+   // See plugin-features.h and preset-features.h
+   //
+   // The feature string is arbitrary, it is the indexer's job to understand it and remap it to its
+   // internal categorization and tagging system.
+   //
+   // Examples:
+   // kick, drum, tom, snare, clap, cymbal, bass, lead, metalic, hardsync, crossmod, acid,
+   // distorted, drone, pad, dirty, etc...
+   void(CLAP_ABI *add_feature)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *feature);
+
+   // Adds extra information to the metadata.
+   void(CLAP_ABI *add_extra_info)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  const char                                           *key,
+                                  const char                                           *value);
```

Comment:
Could this be used e.g. to add information about a preview sample that could be played back when selecting a preset in the host plugin browser?

Reply:
Yes it could, for now there isn't such thing.
I believe that having a preview should be as simple as loading the preset and sending a note or some audio for effects, this would save disk space.